lib/usd/translators/meshWriter.cpp

```diff
@@ -120,7 +129,36 @@ PxrUsdTranslators_MeshWriter::writeMeshAttrs(
     // determine whether we use the "input" or "final" mesh when exporting
     // mesh geometry. This should only be run once at default time.
     if (usdTime.IsDefault()) {
-        _skelInputMesh = writeSkinningData(primSchema);
+        const TfToken& exportSkin = _GetExportArgs().exportSkin;
+        if (exportSkin != UsdMayaJobExportArgsTokens->auto_ &&
+            exportSkin != UsdMayaJobExportArgsTokens->explicit_) {
+        
+            _skelInputMesh = MObject();
+        
+        }else if (exportSkin == UsdMayaJobExportArgsTokens->explicit_ &&
+                  !UsdSkelRoot::Find(primSchema.GetPrim())) {
+        
+            _skelInputMesh = MObject();
+        
+        }else {
+
+            SdfPath skelPath;
+            _skelInputMesh = UsdMayaJointUtil::writeSkinningData(primSchema,
+                                                                 GetUsdPath(),
+                                                                 GetDagPath(),
+                                                                 skelPath,
+                                                                 _GetExportArgs().stripNamespaces, 
+                                                                *_GetSparseValueWriter());
+
+            // Add all skel primvars to the exclude set.
+            // We don't want later processing to stomp on any of our data.
+            _excludeColorSets.insert({_tokens->skelJointIndices,
+                                      _tokens->skelJointWeights,
+                                      _tokens->skelGeomBindTransform});
+
+            // Mark the bindings for post processing.
+            _writeJobCtx.MarkSkelBindings(primSchema.GetPrim().GetPath(), skelPath, exportSkin);
```

Comment:
Looking at the original code, I think this should be skipped if `_skelInputMesh` is a null object coming out of `writeSkinningData()`?

Reply:
Correct, good catch. 

In the original version of writeSkinningData, we were doing early returns if objects were not valid and `_excludeColorSets.insert` and `_writeJobCtx.MarkSkelBindings` would get call right at the end. 

See 2d150798c6f9abc919325d0badf6dd4a185e3381